//
// Copyright 2017 Asylo authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto2";

// $title: Enclave gRPC Server Configuration
// $overview: Configuration options for a basic enclave gRPC server.
// $location: https://asylo.dev/docs/reference/proto/grpc/asylo.enclave_server.v1.html
// $front_matter: redirect_from: /docs/reference/proto/grpc/enclave_server.html

// This package defines messages and extensions used by the EnclaveServer
// utility.
package asylo;

import "asylo/enclave.proto";

// Represents an enclave gRPC server's configuration.
message ServerConfig {
  // The host to run on in the form of IPv6 address minus the port.
  optional string host = 1;

  // The port to run on. A port of 0 indicates that the port should be
  // auto-selected by the system.
  optional int32 port = 2;
}

extend EnclaveConfig {
  // Initial configuration for the gRPC server.
  optional ServerConfig server_input_config = 174238459;
}

extend EnclaveOutput {
  // The gRPC server's current configuration. This may be different than the
  // address provided to the enclave at initialization if port auto-selection
  // was used.
  optional ServerConfig server_output_config = 196851825;
}
